lib/datarockets/style/cop/layout/array_alignment_extended.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+module Datarockets
+  module Style
+    module Cop
+      module Layout
+        # Here we check if the elements of a multi-line array literal are
+        # aligned.
+        #
+        # @example EnforcedStyle: with_first_argument (default)
+        #   # good
+        #
+        #   array = [1, 2, 3,
+        #            4, 5, 6]
+        #   array = ['run',
+        #            'forrest',
+        #            'run']
+        #
+        #   # bad
+        #
+        #   array = [1, 2, 3,
+        #     4, 5, 6]
+        #   array = ['run',
+        #        'forrest',
+        #        'run']
+        #
+        # @example EnforcedStyle: with_fixed_indentation
+        #   # good
+        #
+        #   array = [1, 2, 3,
+        #     4, 5, 6]
+        #
+        #   # bad
+        #
+        #   array = [1, 2, 3,
+        #            4, 5, 6]
+        class ArrayAlignmentExtended < RuboCop::Cop::Cop
+          include RuboCop::Cop::Alignment
+
+          ALIGN_PARAMS_MSG = "Align the elements of an array literal if they span more " \
+            "than one line."
+
+          FIXED_INDENT_MSG = "Use one level of indentation for elements " \
+            "following the first line of a multi-line array."
+
+          def on_array(node)
+            return if node.children.size < 2
+
+            check_alignment(node.children, base_column(node, node.children))
+          end
+
+          def autocorrect(node)
+            RuboCop::Cop::AlignmentCorrector.correct(processed_source, node, column_delta)
+          end
+
+          private
+
+          def message(_node)
+            fixed_indentation? ? FIXED_INDENT_MSG : ALIGN_PARAMS_MSG
+          end
+
+          def fixed_indentation?
+            cop_config["EnforcedStyle"] == "with_fixed_indentation"
+          end
+
+          def base_column(node, args)
+            if fixed_indentation?
+              lineno = target_method_lineno(node)
```

Comment:
let's move all these logic into the separate method so that the method won't be so big